### PR TITLE
Added Drozone to favorites quickaccess

### PR DESCRIPTION
**Pull request by newhinton**
closes #10908
This adds the ability to drop a folder on the quickaccess and add it to it

### CODE REVIEW

**Comment by newhinton** [comment]
CI is failing, but i think the errors are unrelated:
```  
Scenario: create a new user with a custom display name                       # /drone/src/github.com/nextcloud/server/tests/acceptance/features/users.feature:12
    Given I am logged in as the admin                                          # LoginPageContext::iAmLoggedInAsTheAdmin()
    And I open the User settings                                               # SettingsMenuContext::iOpenTheUserSettings()
    When I click the New user button                                           # UsersSettingsContext::iClickTheNewUserButton()
    And I see that the new user form is shown                                  # UsersSettintestgsContext::iSeeThatTheNewUserFormIsShown()
    And I set the user name for the new user to "test"                         # UsersSettingsContext::iSetTheUserNameForTheNewUserTo()
    And I set the display name for the new user to "Test display name"         # UsersSettingsContext::iSetTheDisplayNameForTheNewUserTo()
    And I set the password for the new user to "123456acb"                     # UsersSettingsContext::iSetThePasswordForTheNewUserTo()
    And I create the new user                                                  # UsersSettingsContext::iCreateTheNewUser()
    Then I see that the list of users contains the user "test"                 # UsersSettingsContext::iSeeThatTheListOfUsersContainsTheUser()
    And I see that the display name for the user "test" is "Test display name" # UsersSettingsContext::iSeeThatTheDisplayNameForTheUserIs()
      Row for user test in Users Settings could not be found after 100 seconds
      displayName cell for user test in Users Settings could not be found after 100 seconds
      displayName input for user test in Users Settings could not be found after 100 seconds (NoSuchElementException)
sh: 1: kill: No such process

```

and
```

Enable an app bundle                                          # /drone/src/github.com/nextcloud/server/tests/acceptance/features/apps.feature:42
    Given I act as Jane                                                   # ActorContext::iActAs()
    And I am logged in as the admin                                       # LoginPageContext::iAmLoggedInAsTheAdmin()
    And I open the Apps management                                        # SettingsMenuContext::iOpenTheAppsManagement()
    And I open the "App bundles" section                                  # AppNavigationContext::iOpenTheSection()
    When I enable all apps from the "Enterprise bundle"                   # AppsManagementContext::iEnableAllAppsFromThe()
    Then I see that the "Auditing / Logging" app has been enabled         # AppsManagementContext::iSeeThatTheAppHasBeenEnabled()
      Disable button in the app list for Auditing / Logging could not be found after 100 seconds (NoSuchElementException)
And I see that the "LDAP user and group backend" app has been enabled # AppsManagementContext::iSeeThatTheAppHasBeenEnabled()

```

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -349,10 +349,16 @@
 
 			$.event.trigger({type: "droppedOnTrash"});
 
-			var self=this;
+			var _self=this;
```
Comment (danxuliu) [inline apps/files/js/filelist.js]: Why the rename?

File: apps/files/js/filelist.js [hunk 2]
```diff
@@ -349,10 +349,16 @@
...
 			this.$fileList.on("droppedOnTrash", function (event, filename, directory) {
 				//self.fileActions.triggerAction('Favorite', self.getModelForFile(file), self);
-				self.do_delete(filename, directory)
+				_self.do_delete(filename, directory);
+			});
+
+			$.event.trigger({type: "droppedOnFavorites"});
```
Comment (danxuliu) [inline apps/files/js/filelist.js]: Why is this needed?

File: apps/files/js/filelist.js [hunk 3]
```diff
@@ -349,10 +349,16 @@
...
+			_self=this;
```
Comment (danxuliu) [inline apps/files/js/filelist.js]: `_self` was not modified since it was set, so it is not needed to set it again.

**Review by danxuliu — COMMENTED** [review]
Please rebase on current master :-) (then you will need to force push the branch again).

Tested and works, but it should be tested again once rebased ;-)

Maybe it would be good to not enable the favorite drop zone when dragging an already favorited folder, but that is a minor detail. In any case, a really nice feature, thanks :-D

**Comment by danxuliu** [comment]
>  CI is failing, but i think the errors are unrelated:

Yes, `sh: 1: kill: No such process` is a crash in the PHP built-in server (do not worry, you did not cause it :-P ), and the other failure was caused by an app not having been released yet in the app store when the tests were run (again, nothing that you need to worry about ;-) ).

File: apps/files/js/filelist.js (hunk 1 above)
Comment (newhinton) [reply apps/files/js/filelist.js]: Initially i had problems with self, it would not work properly. I can look if it works now

File: apps/files/js/filelist.js (hunk 3 above)
Comment (newhinton) [reply apps/files/js/filelist.js]: You are right, this is a leftover because i implemented the trash dropzone and the favorites dropzone seperately and needed to make them seperate

File: apps/files/js/filelist.js (hunk 2 above)
Comment (newhinton) [reply apps/files/js/filelist.js]: This deletes the folder when it is dropped on the trash

**Comment by danxuliu** [comment]
@newhinton, sorry for the delay.

I have taken the liberty of rebasing again your pull request and do some cleaning of the drop zone feature and the pull request history. Please refer to each individual commit and its comment for further information.

Regarding the commit that adds the drop zone to the favorites I have removed `.attr('data-favorite', elem.tags.includes("_$!<Favorite>!$_"))` (which should have been `.attr('data-favorite', elem.tags.includes(OC.TAG_FAVORITES))` ;-) ) because it was not needed with the current handling of the drop on the favorites and because, due to bugs in other parts of the Files code, `elem` may not always have a `tags` attribute (even if it should), which broke dragging rows in the file list (this could happen, for example, when using the _Select all_ checkbox).

**Review by danxuliu — APPROVED** [review]
Tested and works :+1:

Currently favoriting by dropping is limited to only one file or folder at a time due to some unrelated bugs in the Files app; I would merge this anyway in its current state.

Thanks again @newhinton!

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by newhinton** [comment]
I didn't know that there was an OC.Tag introduced, i wrote a majority of this back at the conf, so there may have been some changes which im not aware of ^^ but good to know :D

**Comment by danxuliu** [comment]
No problem at all about that @newhinton, you did not have to know it; I just mentioned it for your information :-)
